sdks/python/apache_beam/tools/coders_microbenchmark.py

```diff
@@ -0,0 +1,141 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A microbenchmark for measuring performance of coders.
+
+This runs a sequence of encode-decode operations on random inputs
+to collect a per-element performance of various coders.
+
+Run as
+  python -m apache_beam.tools.coders_microbenchmark
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import collections
+import random
+import string
+import sys
+
+from apache_beam.coders import coders
+from apache_beam.tools import utils
+from apache_beam.transforms import window
+from apache_beam.utils import windowed_value
+
+
+def coder_benchmark_factory(name, coder, generate_fn):
+  def init(self, size):
+    self.coder_ = coder
+    self.data_ = generate_fn(size)
+
+  def call(self):
+    _ = self.coder_.decode(self.coder_.encode(self.data_))
```

Comment:
We care more about the speed of encoding a list of 10000 small dicts, than a single dict of 10000 entries. The former better approximates how the code will get invoked on PCollections. 

E.g. you could have 

```
def coder_benchmark(name, coder, *sample_elements):
  def __init__(self, size):
    self._coder = IterableCoder(coder)
    self._elements = list(itertools.islice(itertools.cycle(sample_elements), size))
```

Reply:
Changed. Let me know if that captures your idea. Currently benchmarks use a list of 1000 collections of size 50. The suite takes around 2 min to complete on my machine. Dict[str, int] and List of WindowedValues benchmarks take 10x longer to finish than others. With 10000 collections, the  execution time might be a little longer than desirable for a quick feedback loop, but we can increase it if necessary.